yetanotherco/aligned_layer
# feat: Telemetry for the batcher and improve current tracing

# Telemetry integration for the Batcher

## Description

This PR includes the telemetry integration on the batcher side to log the start of a task and some extra steps between adding more visibility on the different parts that process a given batch.
Now we include what are called `subspans` that are intermediate logs that will pop up instantly on the Jaeger dashboard, now we are able to track on which part of the process a task failed or get lost.

## How to test

- Setup all the aligned components (anvil, aggregator, operator and batcher)
- Run the telemetry module
   - `make telemetry_create_env`
   - `make telemetry_run_db`
   - `make open_telemetry_start`
   - `make telemetry_start`
- Send proofs using `make batcher_send_burst_groth16`
- Open the Jaeger dashboard in http://localhost:16686
- You should see new logs for the batcher like:
<img width="1488" alt="image" src="https://github.com/user-attachments/assets/e9c0bb62-f5db-4c2a-a463-07d22baf4796">
- Also you should see new logs for the aggregator
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/405bb2b1-de36-49f1-9749-a0ee03d6371e">

You can take down the operator to check that also the expired tasks should be logged in the aggregator span after some minutes.

## Type of change


## Checklist

  - [Link to the open issue addressing it]()
